lib/ignored-paths.js

```diff
@@ -92,14 +92,24 @@ class IgnoredPaths {
         }
 
         /**
+         * read ignore filepath
+         * @param {string} filepath, file to add to ig
+         * @returns {array} raw ignore rules
+         */
+        function readIgnoreFile(filepath) {
+            return fs.readFileSync(filepath, "utf8");
+        }
```

Comment:
@not-an-aardvark 
you're right. For some reason, the method approach was showing errors before, which is why I had the previous approach with passing the cache as a parameter. I am not sure why I had errors at that point.

I moved addIgnoreFile and readIgnoreFile to class methods now and it is much simpler.

Added a test as well, counting that the calls to readFileSync only happens once in the test instead of the two-time reads that happens right now when `addIgnoreFile` gets called twice in succession.